src/flattenDeep.ts

```diff
@@ -12,19 +12,24 @@ type FlattenDeep4<T> = T extends ReadonlyArray<infer K> ? K : T;
 /**
  * Recursively flattens `array`.
  *
+ * ! **DEPRECATED** Use `R.flat(data, 4)`. Will be removed in V2!
+ *
  * @param items - The target array.
  * @signature
  *   R.flattenDeep(array)
  * @example
  *    R.flattenDeep([[1, 2], [[3], [4, 5]]]) // => [1, 2, 3, 4, 5]
  * @pipeable
  * @category Array
+ * @deprecated Use `R.flat(data, 4)`. Will be removed in V2!
```

Comment:
Notice that the built-in type for `Array.prototype.flat` is broken and doesn't result in a usable type for the output array. Even if MDN suggest it, it looks like the Typescript team built the type assuming it would be used with a literal number up to 20. I'm trying to rewrite the type to support arbitrary depths up to infinity and to support tuples more naturally. It ain't easy though...

Reply:
if the typescript definitions themselves dont support arbitrary depth, i dont see much point in trying to go higher